src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneIndexingService.cs

```diff
@@ -0,0 +1,209 @@
+                {
+                    return;
+                }
+            }
+            else {
+                var taskId = _indexingState.GetLastTaskId(indexName);
+                lastTaskId = Math.Min(lastTaskId, taskId);
+                allIndices.Add(indexName, taskId);
+            }
+
+            var batch = Array.Empty<IndexingTask>();
+
+            do
+            {
+                // Create a scope for the content manager
+                var shellScope = await _shellHost.GetScopeAsync(_shellSettings);
+
+                await shellScope.UsingAsync(async scope =>
+                {
+                    // Load the next batch of tasks
+                    batch = (await _indexingTaskManager.GetIndexingTasksAsync(lastTaskId, BatchSize)).ToArray();
+
+                    if (!batch.Any())
+                    {
+                        return;
+                    }
+
+                    foreach (var task in batch)
+                    {
+                        var contentManager = scope.ServiceProvider.GetRequiredService<IContentManager>();
+                        var indexHandlers = scope.ServiceProvider.GetServices<IContentItemIndexHandler>();
+
+                        foreach (var index in allIndices)
+                        {
+                            // TODO: ignore if this index is not configured for the content type
+
+                            if (index.Value < task.Id)
+                            {
+                                _indexManager.DeleteDocuments(index.Key, new string[] { task.ContentItemId });
+                            }
+                        }
+
+                        if (task.Type == IndexingTaskTypes.Update)
+                        {
+                            var contentItem = await contentManager.GetAsync(task.ContentItemId);
+
+                            if (contentItem == null)
+                            {
+                                continue;
+                            }
+
+                            var context = new BuildIndexContext(new DocumentIndex(task.ContentItemId), contentItem, new string[] { contentItem.ContentType });
+
+                            // Update the document from the index if its lastIndexId is smaller than the current task id. 
+                            await indexHandlers.InvokeAsync(x => x.BuildIndexAsync(context), Logger);
+
+                            foreach (var index in allIndices)
+                            {
+                                if (index.Value < task.Id)
+                                {
+                                    _indexManager.StoreDocuments(index.Key, new DocumentIndex[] { context.DocumentIndex });
+                                }
+                            }
+                        }
+                    }
+
+                    // Update task ids
+                    lastTaskId = batch.Last().Id;
+
+                    foreach (var index in allIndices)
```

Comment:
Here it will set the LastTaskId of all indices once it completed it's task of indexing all content items.